### PR TITLE
revsets.md: document the `all:` prefix

### PR DESCRIPTION
**Pull request by ilyagr**
While this is arguably not part of the revset language, this is a likely place for a user to look.

See https://discord.com/channels/968932220549103686/968932220549103689/1228065431281995837

# Checklist

If applicable:
- [ ] I have updated `CHANGELOG.md`
- [x] I have updated the documentation (README.md, docs/, demos/)
- [ ] I have updated the config schema (cli/src/config-schema.json)
- [ ] I have added tests to cover my changes

### CODE REVIEW

File: docs/revsets.md [hunk 1]
```diff
@@ -70,6 +70,29 @@ only symbols.
...
+`xyz` as long as `xyz` has exactly one child.
+
+If `xyz` has more than one child, without the `all:` prefix, `jj rebase -r w -d
+xyz+` will return an error.
+
+With the `all:` prefix and if if `xyz` has more than one child, `jj rebase -r w
+-d all:xyz+` will make `w` into a merge commit. The `all:` prefix confirms that
+the user expected `xyz` to have more than one child.
+
+A more useful example: if `w` is a merge commit, `jj rebase -s w -d all:w- -d
```
Comment (ilyagr) [inline docs/revsets.md:92]: TODO suggested by @bnjmnt4n on Discord:


> is it worth leaving a note about the new option from https://github.com/martinvonz/jj/pull/3458 as well?

The option is `ui.always-allow-large-revsets`

Comment (ilyagr) [reply docs/revsets.md:99]: PTAL. I moved most of the docs to `revsets.md`.

File: docs/revsets.md [hunk 2]
```diff
@@ -70,6 +70,38 @@ only symbols.
 You can use parentheses to control evaluation order, such as `(x & y) | z` or
 `x & (y | z)`.
 
+### The `all:` prefix
```
Comment (yuja) [inline docs/revsets.md:73]: nit: I would move this section around "Aliases" or "Examples". As you mentioned, it's not technically a language construct, and can't be inserted in the middle of expressions.

Comment (yuja) [inline docs/revsets.md:73]: nit: I use "modifier" instead of "prefix" in the grammar and error message because I needed some terminology for that.

Comment (ilyagr) [reply docs/revsets.md:73]: Thanks! I'll adjust it. 

I'm also wondering whether "prefix" is a better name, but that's beyond the scope of this PR.

Comment (ilyagr) [reply docs/revsets.md:73]: I'm unsure what's best, though either way seems OK. It *looks*  like a language construct but isn't.

The interaction that caused this PR was somebody asking "what is the `:` operator when they saw `jj rebase -s all:...` suggested in a forum. For this situation, placing it next to the language construct seems better. On the other hand, perhaps we shouldn't over-optimize for this one anecdote.

File: docs/revsets.md [hunk 3]
```diff
@@ -70,6 +70,38 @@ only symbols.
...
+
+An `all:` prefix before a revset expression does not change its meansing.
```
Comment (thoughtpolice) [inline docs/revsets.md:75]: `s/meansing/meaning/`

File: docs/revsets.md (hunk 1 above)
Comment (thoughtpolice) [reply docs/revsets.md:101]: The move LGTM!

File: docs/revsets.md [hunk 4]
```diff
@@ -70,6 +70,38 @@ only symbols.
...
+Strictly speaking, it is not part of the revset language. The notation is
+similar to the prefixes allowed before [string patterms](#string-patterns).
+
+Certain commands (such as `jj rebase`) can take multiple revset arguments, and
+each of these may resolve to one-or-many revisions. By default, `jj` will not
+allow revsets that resolve to more than one revision &mdash; a so-called "large
+revset" &mdash; and will ask you to confirm that you want to proceed by
+prefixing it with the `all:` modifier.
```
Comment (thoughtpolice) [inline docs/revsets.md:83]: This paragraph should probably go first, before the proceeding two? And the first two paragraphs can just be collapsed into one.

File: docs/revsets.md (hunk 2 above)
Comment (ilyagr) [reply docs/revsets.md]: I moved the section down as you requested. It should be fine. If enough people seem to be unable to find it when they need it, I might be tempted to move it up again.

File: docs/revsets.md (hunk 4 above)
Comment (ilyagr) [reply docs/revsets.md]: Done, thanks!

File: docs/config.md [hunk 5]
```diff
@@ -328,24 +328,16 @@ Can be customized by the `format_short_signature()` template alias.
 
 ### Allow "large" revsets by default
 
-Certain commands (such as `jj rebase`) can take multiple revset arguments, and
-each of these may resolve to one-or-many revisions. By default, `jj` will not
-allow revsets that resolve to more than one revision &mdash; a so-called "large
-revset" &mdash; and will ask you to confirm that you want to proceed by
-prefixing it with the `all:` modifier.
+By default, `jj` requires the `all:` prefix for many commands to accept a revset
```
Comment (yuja) [inline docs/config.md:331]: nit: "many commands" seems wrong. Maybe reuse the original explanation?

> Certain commands (such as `jj rebase`) can take multiple revset arguments, but require the `all:` prefix ...

**Review by yuja — APPROVED** [review]

File: docs/config.md (hunk 5 above)
Comment (ilyagr) [reply docs/config.md:331]: Good point, thanks! I also needed to replace "prefix" with "modifier" here.

Let me know if you think this needs more changes.

**Pull request merged**